### DISCUSSION TITLE
DMARC records are not generated correctly for non-primary domains

### DISCUSSION THREAD

**Discussion by isdn** [Mailu/Mailu#3082] [User support] [answered]
Mailu version 2.0.34

DMARC records for non-primary domains refer to the primary domain.

For example:
- primary domain (set up during installation and initial configuration): example.com
- another domain (added later via admin UI): example1.com

Generated DNS DMARC entry for example1.com is the following:
```
_dmarc.example1.com. 600 IN TXT "v=DMARC1; p=reject; rua=mailto:admin@example.com; ruf=mailto:admin@example.com; adkim=s; aspf=s"
example1.com._report._dmarc.example.com. 600 IN TXT "v=DMARC1"
```
That entry produces the following error at https://mxtoolbox.com/dmarc.aspx?source=dmarctools-content :

```
DMARC External Validation 
External Domains in your DMARC are not giving permission for your reports to be sent to them. 
```

As far as I understand, it should be the following in this case:

```
_dmarc.example1.com. 600 IN TXT "v=DMARC1; p=reject; rua=mailto:admin@example1.com; ruf=mailto:admin@example1.com; adkim=s; aspf=s"
example1.com._report._dmarc.example1.com. 600 IN TXT "v=DMARC1"
```

Thanks!

**Comment by isdn** [comment]
Hi!

That is not a "User support" issue. I already corrected the DNS records and everything seems to be good.

**Reply by ghostwheel42** [reply, marked as answer]
I've implemented this as it is now, because I wanted to use a single domain or postmaster address to gather all reports.
You just have to add the second record on example.com to make this work:

`example1.com._report._dmarc.example.com. 600 IN TXT "v=DMARC1"
`

The second record is displayed in master. If it is not shown in the current version it will be included in the next release. 

Alex

**Reply by isdn** [reply]
Ok, thank you.
